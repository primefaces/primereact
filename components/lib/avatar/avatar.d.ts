/**
 *
 * Avatar represents people using icons, labels and images.
 *
 * [Live Demo](https://www.primereact.org/avatar)
 *
 * @module avatar
 *
 */
import * as React from 'react';
import { IconType, PassThroughType } from '../utils';

export declare type AvatarPassThroughType<T> = PassThroughType<T, AvatarPassThroughMethodOptions>;
/**
 * Custom passthrough(pt) option method.
 */
export interface AvatarPassThroughMethodOptions {
    props: AvatarProps;
}

/**
 * Custom passthrough(pt) options.
 * @see {@link AvatarProps.pt}
 */
export interface AvatarPassThroughOptions {
    /**
     * Uses to pass attributes to the root's DOM element.
     */
    root?: AvatarPassThroughType<React.HTMLAttributes<HTMLDivElement>>;
    /**
     * Uses to pass attributes to the label's DOM element.
     */
    label?: AvatarPassThroughType<React.HTMLAttributes<HTMLSpanElement>>;
    /**
     * Uses to pass attributes to the icon's DOM element.
     */
    icon?: AvatarPassThroughType<React.SVGProps<SVGSVGElement> | React.HTMLAttributes<HTMLSpanElement>>;
    /**
     * Uses to pass attributes to the image's DOM element.
     */
    image?: AvatarPassThroughType<React.ImgHTMLAttributes<HTMLImageElement>>;
}

/**
 * Defines valid properties in Avatar component. In addition to these, all properties of HTMLDivElement can be used in this component.
 * @group Properties
 */
export interface AvatarProps extends Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, 'ref'> {
    /**
     * Defines the icon to display.
     */
    icon?: IconType<AvatarProps> | undefined;
    /**
     * Defines the image to display.
     */
    image?: string | undefined;
    /**
     * It specifies an alternate text for an image, if the image cannot be displayed.
     * @defaultValue avatar
     */
    imageAlt?: string | undefined;
    /**
     * Defines a fallback image or URL if the main image fails to load. If "default" will fallback to label then icon.
     * @defaultValue default
     */
    imageFallback?: 'default' | string | undefined;
    /**
     * Defines the text to display.
     */
    label?: string | undefined;
    /**
     * Shape of the element.
     * @defaultValue square
     */
    shape?: 'square' | 'circle' | undefined;
    /**
     * Size of the element.
     * @defaultValue normal
     */
    size?: 'normal' | 'large' | 'xlarge' | undefined;
    /**
     * Template of the content.
     */
    template?: React.ReactNode | ((props: AvatarProps) => React.ReactNode);
    /**
     * This event is triggered if an error occurs while loading an image file.
     * @param {React.SyntheticEvent}  event - Browser event.
     */
    onImageError?(event: React.SyntheticEvent): void;
    /**
     * Callback to invoke on click.
     * @param {React.MouseEvent<HTMLElement>} event - Browser event.
     */
    onClick?(event: React.MouseEvent<HTMLElement>): void;
    /**
     * Used to get the child elements of the component.
     * @readonly
     */
    children?: React.ReactNode | undefined;
    /**
     * Uses to pass attributes to DOM elements inside the component.
     * @type {AvatarPassThroughOptions}
     */
    pt?: AvatarPassThroughOptions;
}

/**
 * **PrimeReact - Avatar**
 *
 * _Avatar represents people using icons, labels and images._
 *
 * [Live Demo](https://www.primereact.org/avatar/)
 * --- ---
 * ![PrimeReact](https://primefaces.org/cdn/primereact/images/logo-100.png)
 *
 * @group Component
 */
export declare class Avatar extends React.Component<AvatarProps, any> {
    /**
     * Used to get container element.
     * @return {HTMLDivElement} Container element
     */
    public getElement(): HTMLDivElement;
}
