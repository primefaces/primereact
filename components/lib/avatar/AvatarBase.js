import { ObjectUtils } from '../utils/Utils';

export const AvatarBase = {
    defaultProps: {
        __TYPE: 'Avatar',
        className: null,
        icon: null,
        image: null,
        imageAlt: 'avatar',
        imageFallback: 'default',
        label: null,
        onImageError: null,
        shape: 'square',
        size: 'normal',
        style: null,
        template: null,
        children: undefined
    },
    getProps: (props) => ObjectUtils.getMergedProps(props, AvatarBase.defaultProps),
    getOtherProps: (props) => ObjectUtils.getDiffProps(props, AvatarBase.defaultProps)
};
