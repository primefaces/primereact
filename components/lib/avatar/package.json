{
    "main": "./avatar.cjs.js",
    "module": "./avatar.esm.js",
    "unpkg": "./avatar.min.js",
    "types": "./avatar.d.ts",
    "sideEffects": false
}
