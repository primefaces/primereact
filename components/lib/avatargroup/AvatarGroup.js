import * as React from 'react';
import { classNames, ObjectUtils } from '../utils/Utils';

export const AvatarGroup = React.forwardRef((props, ref) => {
    const otherProps = ObjectUtils.findDiffKeys(props, AvatarGroup.defaultProps);
    const className = classNames('p-avatar-group p-component', props.className);

    return (
        <div className={className} style={props.style} {...otherProps}>
            {props.children}
        </div>
    )
});

AvatarGroup.displayName = 'AvatarGroup';
AvatarGroup.defaultProps = {
    __TYPE: 'AvatarGroup',
    style: null,
    className: null
}
