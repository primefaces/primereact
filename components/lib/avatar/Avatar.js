import * as React from 'react';
import { classNames, IconUtils, ObjectUtils } from '../utils/Utils';
import { AvatarBase } from './AvatarBase';

export const Avatar = React.forwardRef((inProps, ref) => {
    const props = AvatarBase.getProps(inProps);

    const elementRef = React.useRef(null);
    const [imageFailed, setImageFailed] = React.useState(false);

    const createContent = () => {
        if (props.image && !imageFailed) {
            return <img src={props.image} alt={props.imageAlt} onError={onImageError}></img>;
        } else if (props.label) {
            return <span className="p-avatar-text">{props.label}</span>;
        } else if (props.icon) {
            return IconUtils.getJSXIcon(props.icon, { className: 'p-avatar-icon' }, { props });
        }

        return null;
    };

    const onImageError = (event) => {
        if (props.imageFallback === 'default') {
            if (!props.onImageError) {
                // fallback to label or icon
                setImageFailed(true);
                event.target.src = null;
            }
        } else {
            // try fallback as an image
            event.target.src = props.imageFallback;
        }

        props.onImageError && props.onImageError(event);
    };

    React.useImperativeHandle(ref, () => ({
        props,
        getElement: () => elementRef.current
    }));

    const otherProps = AvatarBase.getOtherProps(props);
    const containerClassName = classNames(
        'p-avatar p-component',
        {
            'p-avatar-image': props.image !== null && !imageFailed,
            'p-avatar-circle': props.shape === 'circle',
            'p-avatar-lg': props.size === 'large',
            'p-avatar-xl': props.size === 'xlarge',
            'p-avatar-clickable': !!props.onClick
        },
        props.className
    );

    const content = props.template ? ObjectUtils.getJSXElement(props.template, props) : createContent();

    return (
        <div ref={elementRef} className={containerClassName} style={props.style} {...otherProps}>
            {content}
            {props.children}
        </div>
    );
});

Avatar.displayName = 'Avatar';
