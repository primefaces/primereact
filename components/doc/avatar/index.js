import React, { memo } from 'react';
import Link from 'next/link';
import { TabView, TabPanel } from '../../lib/tabview/TabView';
import { useLiveEditorTabs } from '../common/liveeditor';
import { CodeHighlight } from '../common/codehighlight';

const AvatarDoc = memo(() => {

    const sources = {
        'class': {
            tabName: 'Class Source',
            content: `
import React, { Component } from 'react';
import { Avatar } from 'primereact/avatar';
import { AvatarGroup } from 'primereact/avatargroup';
import { Badge } from 'primereact/badge';

export class AvatarDemo extends Component {

    render() {
        return (
            <div>
                <div className="grid">
                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Label</h5>
                            <Avatar label="P" className="mr-2" size="xlarge" />
                            <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                            <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Label - Circle</h5>
                            <Avatar label="P" className="mr-2" size="xlarge" shape="circle" />
                            <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                            <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Label - Badge</h5>
                            <Avatar label="U" size="xlarge" className="p-overlay-badge" style={{ backgroundColor: '#4caf4f', color: '#ffffff' }}>
                                <Badge value="4" />
                            </Avatar>
                        </div>
                    </div>
                </div>

                <div className="grid">
                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Icon</h5>
                            <Avatar icon="pi pi-user" className="mr-2" size="xlarge" />
                            <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                            <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Icon - Circle</h5>
                            <Avatar icon="pi pi-user" className="mr-2" size="xlarge" shape="circle" />
                            <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                            <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Icon - Badge</h5>
                            <Avatar className="p-overlay-badge" icon="pi pi-user" size="xlarge" ><Badge value="4" /> </Avatar>
                        </div>
                    </div>
                </div>

                <div className="grid">
                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Image</h5>
                            <Avatar image="images/avatar/amyelsner.png" className="mr-2" size="xlarge" shape="circle" />
                            <Avatar image="images/avatar/asiyajavayant.png" className="mr-2" size="large" shape="circle" />
                            <Avatar image="images/avatar/onyamalimba.png" className="mr-2" shape="circle" />
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Avatar Group</h5>
                            <AvatarGroup className="mb-3">
                                <Avatar image="images/avatar/amyelsner.png" size="large" shape="circle" />
                                <Avatar image="images/avatar/asiyajavayant.png" size="large" shape="circle" />
                                <Avatar image="images/avatar/onyamalimba.png" size="large" shape="circle" />
                                <Avatar image="images/avatar/ionibowcher.png" size="large" shape="circle" />
                                <Avatar image="images/avatar/xuxuefeng.png" size="large" shape="circle" />
                                <Avatar label="+2" shape="circle" size="large" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                            </AvatarGroup>
                        </div>
                    </div>

                    <div className="col-12 md:col-4">
                        <div className="card">
                            <h5>Image - Badge</h5>
                            <Avatar className="p-overlay-badge" image="demo/images/organization/walter.jpg" size="xlarge">
                                <Badge value="4" severity="danger" />
                            </Avatar>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
`
        },
        'hooks': {
            tabName: 'Hooks Source',
            content: `
import React from 'react';
import { Avatar } from 'primereact/avatar';
import { AvatarGroup } from 'primereact/avatargroup';
import { Badge } from 'primereact/badge';

export const AvatarDemo = () => {

    return (
        <div>
            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Circle</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Badge</h5>
                        <Avatar label="U" size="xlarge" className="p-overlay-badge" style={{ backgroundColor: '#4caf4f', color: '#ffffff' }}>
                            <Badge value="4" />
                        </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Circle</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Badge</h5>
                        <Avatar className="p-overlay-badge" icon="pi pi-user" size="xlarge" ><Badge value="4" /> </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image</h5>
                        <Avatar image="images/avatar/amyelsner.png" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar image="images/avatar/asiyajavayant.png" className="mr-2" size="large" shape="circle" />
                        <Avatar image="images/avatar/onyamalimba.png" className="mr-2" shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Avatar Group</h5>
                        <AvatarGroup className="mb-3">
                            <Avatar image="images/avatar/amyelsner.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/asiyajavayant.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/onyamalimba.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/ionibowcher.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/xuxuefeng.png" size="large" shape="circle" />
                            <Avatar label="+2" shape="circle" size="large" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                        </AvatarGroup>
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image - Badge</h5>
                        <Avatar className="p-overlay-badge" image="images/organization/walter.jpg" size="xlarge">
                            <Badge value="4" severity="danger" />
                        </Avatar>
                    </div>
                </div>
            </div>
        </div>
    );
}
`
        },
        'ts': {
            tabName: 'TS Source',
            content: `
import React from 'react';
import { Avatar } from 'primereact/avatar';
import { AvatarGroup } from 'primereact/avatargroup';
import { Badge } from 'primereact/badge';

export const AvatarDemo = () => {

    return (
        <div>
            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Circle</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Badge</h5>
                        <Avatar label="U" size="xlarge" className="p-overlay-badge" style={{ backgroundColor: '#4caf4f', color: '#ffffff' }}>
                            <Badge value="4" />
                        </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Circle</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Badge</h5>
                        <Avatar className="p-overlay-badge" icon="pi pi-user" size="xlarge" ><Badge value="4" /> </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image</h5>
                        <Avatar image="images/avatar/amyelsner.png" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar image="images/avatar/asiyajavayant.png" className="mr-2" size="large" shape="circle" />
                        <Avatar image="images/avatar/onyamalimba.png" className="mr-2" shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Avatar Group</h5>
                        <AvatarGroup className="mb-3">
                            <Avatar image="images/avatar/amyelsner.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/asiyajavayant.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/onyamalimba.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/ionibowcher.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/xuxuefeng.png" size="large" shape="circle" />
                            <Avatar label="+2" shape="circle" size="large" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                        </AvatarGroup>
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image - Badge</h5>
                        <Avatar className="p-overlay-badge" image="images/organization/walter.jpg" size="xlarge">
                            <Badge value="4" severity="danger" />
                        </Avatar>
                    </div>
                </div>
            </div>
        </div>
    );
}
`
        },
        'browser': {
            tabName: 'Browser Source',
            imports: `
        <script src="https://unpkg.com/primereact/core/core.min.js"></script>
        <script src="https://unpkg.com/primereact/avatar/avatar.min.js"></script>
        <script src="https://unpkg.com/primereact/avatargroup/avatargroup.min.js"></script>
        <script src="https://unpkg.com/primereact/badge/badge.min.js"></script>`,
            content: `
const { useState, useRef } = React;
const { Avatar } = primereact.avatar;
const { AvatarGroup } = primereact.avatargroup;
const { Badge } = primereact.badge;

const AvatarDemo = () => {

    return (
        <div>
            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Circle</h5>
                        <Avatar label="P" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar label="V" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar label="U" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Label - Badge</h5>
                        <Avatar label="U" size="xlarge" className="p-overlay-badge" style={{ backgroundColor: '#4caf4f', color: '#ffffff' }}>
                            <Badge value="4" />
                        </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Circle</h5>
                        <Avatar icon="pi pi-user" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" size="large" style={{ backgroundColor: '#2196F3', color: '#ffffff' }} shape="circle" />
                        <Avatar icon="pi pi-user" className="mr-2" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Icon - Badge</h5>
                        <Avatar className="p-overlay-badge" icon="pi pi-user" size="xlarge" ><Badge value="4" /> </Avatar>
                    </div>
                </div>
            </div>

            <div className="grid">
                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image</h5>
                        <Avatar image="images/avatar/amyelsner.png" className="mr-2" size="xlarge" shape="circle" />
                        <Avatar image="images/avatar/asiyajavayant.png" className="mr-2" size="large" shape="circle" />
                        <Avatar image="images/avatar/onyamalimba.png" className="mr-2" shape="circle" />
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Avatar Group</h5>
                        <AvatarGroup className="mb-3">
                            <Avatar image="images/avatar/amyelsner.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/asiyajavayant.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/onyamalimba.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/ionibowcher.png" size="large" shape="circle" />
                            <Avatar image="images/avatar/xuxuefeng.png" size="large" shape="circle" />
                            <Avatar label="+2" shape="circle" size="large" style={{ backgroundColor: '#9c27b0', color: '#ffffff' }} />
                        </AvatarGroup>
                    </div>
                </div>

                <div className="col-12 md:col-4">
                    <div className="card">
                        <h5>Image - Badge</h5>
                        <Avatar className="p-overlay-badge" image="images/organization/walter.jpg" size="xlarge">
                            <Badge value="4" severity="danger" />
                        </Avatar>
                    </div>
                </div>
            </div>
        </div>
    );
}
                `
        }
    }

    return (
        <div className="content-section documentation" id="app-doc">
            <TabView>
                <TabPanel header="Documentation">
                    <h5>Import via Module</h5>
<CodeHighlight lang="js">
{`
import { Avatar } from 'primereact/avatar';
import { AvatarGroup } from 'primereact/avatargroup';
`}
</CodeHighlight>

                    <h5>Import via CDN</h5>
<CodeHighlight>
{`
<script src="https://unpkg.com/primereact/core/core.min.js"></script>
<script src="https://unpkg.com/primereact/avatar/avatar.min.js"></script>
<script src="https://unpkg.com/primereact/avatargroup/avatargroup.min.js"></script>
`}
</CodeHighlight>

                    <h5>Getting Started</h5>
                    <p>Avatar has three built-in display modes; "label", "icon" and "image".</p>
<CodeHighlight>
{`
<Avatar label="P" />
<Avatar icon="pi pi-search" />
<Avatar image="user.png" />
`}
</CodeHighlight>
                    <h5>Sizes</h5>
                    <p><i>size</i> property defines the size of the Avatar with "large" and "xlarge" as possible values.</p>
<CodeHighlight>
{`
<Avatar label="P" size="large"/>
`}
</CodeHighlight>
                    <h5>AvatarGroup</h5>
                    <p>A set of Avatars can be displayed together using the <i>AvatarGroup</i> component.</p>
<CodeHighlight>
{`
<AvatarGroup>
    <Avatar label="P" />
    <Avatar icon="pi pi-search" />
    <Avatar image="user.png" />
    <Avatar label="+2" />
</AvatarGroup>
`}
</CodeHighlight>
                    <h5>Shape</h5>
                    <p>Avatar comes in two different styles specified with the <i>shape</i> property, "square" is the default and "circle" is the alternative.</p>
<CodeHighlight>
{`
<Avatar label="P" shape="circle"/>
`}
</CodeHighlight>
                    <h5>Badge</h5>
                    <p>A badge can be added to an Avatar with the <Link href="/badge">Badge</Link> component.</p>
<CodeHighlight>
{`
<Avatar image="user.png" size="xlarge">
    <Badge value="4" severity="danger" />
</Avatar>
`}
</CodeHighlight>
                    <h5>Templating</h5>
                    <p>Content can easily be customized with the default slot instead of using the built-in modes.</p>
<CodeHighlight>
{`
<Avatar>
   Content
</Avatar>
`}
</CodeHighlight>

                    <h5>Properties of Avatar</h5>
                    <p>Any property as style and class are passed to the main container element. Following are the additional properties to configure the component.</p>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                                <tr>
                                    <th>Name</th>
                                    <th>Type</th>
                                    <th>Default</th>
                                    <th>Description</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>label</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Defines the text to display.</td>
                                </tr>
                                <tr>
                                    <td>icon</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Defines the icon to display.</td>
                                </tr>
                                <tr>
                                    <td>image</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Defines the image to display.</td>
                                </tr>
                                <tr>
                                    <td>size</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Size of the element, valid options are "large" and "xlarge".</td>
                                </tr>
                                <tr>
                                    <td>shape</td>
                                    <td>string</td>
                                    <td>square</td>
                                    <td>Shape of the element, valid options are "square" and "circle".</td>
                                </tr>
                                <tr>
                                    <td>template</td>
                                    <td>any</td>
                                    <td>null</td>
                                    <td>Template of the content.</td>
                                </tr>
                                <tr>
                                    <td>imageAlt</td>
                                    <td>any</td>
                                    <td>null</td>
                                    <td>It specifies an alternate text for an image, if the image cannot be displayed.</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Properties of AvatarGroup</h5>
                    <p>Any property as style and class are passed to the main container element. There are no additional properties.</p>

                    <h5>Events</h5>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                                <tr>
                                    <th>Name</th>
                                    <th>Parameters</th>
                                    <th>Description</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>onImageError</td>
                                    <td>event: Browser event</td>
                                    <td>This event is triggered if an error occurs while loading an image file.</td>
                                </tr>
                                <tr>
                                    <td>onClick</td>
                                    <td>event: Browser event</td>
                                    <td>Callback to invoke on click.</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Styling of Avatar</h5>
                    <p>Following is the list of structural style classes, for theming classes visit <Link href="/theming"> theming</Link> page.</p>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                                <tr>
                                    <th>Name</th>
                                    <th>Element</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>p-avatar</td>
                                    <td>Container element.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-image</td>
                                    <td>Container element in image mode.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-circle</td>
                                    <td>Container element with a circle shape.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-text</td>
                                    <td>Text of the Avatar.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-icon</td>
                                    <td>Icon of the Avatar.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-lg</td>
                                    <td>Container element with a large size.</td>
                                </tr>
                                <tr>
                                    <td>p-avatar-xl</td>
                                    <td>Container element with an xlarge size.</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Styling of AvatarGroup</h5>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                                <tr>
                                    <th>Name</th>
                                    <th>Element</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>p-avatar-group</td>
                                    <td>Container element.</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Dependencies</h5>
                    <p>None.</p>
                </TabPanel>

                {
                    useLiveEditorTabs({ name: 'AvatarDemo', sources: sources })
                }
            </TabView>
        </div>
    );
})

export default AvatarDoc;
