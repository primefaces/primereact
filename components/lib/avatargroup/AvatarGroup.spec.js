import '@testing-library/jest-dom';
import { Avatar } from '../avatar/Avatar';
import { AvatarGroup } from './AvatarGroup';

import { snapshot } from '../../test';

describe('AvatarGroup', () => {
    snapshot(<AvatarGroup />, 'default');
    snapshot(
        <AvatarGroup>
            <Avatar label="P" />
            <Avatar icon="pi pi-search" />
            <Avatar image="user.png" />
            <Avatar label="+2" />
        </AvatarGroup>,
        'group'
    );
});
