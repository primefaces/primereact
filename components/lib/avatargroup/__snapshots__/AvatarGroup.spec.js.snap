// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup default 1`] = `
<div>
  <div
    class="p-avatar-group p-component"
  />
</div>
`;

exports[`AvatarGroup group 1`] = `
<div>
  <div
    class="p-avatar-group p-component"
  >
    <div
      class="p-avatar p-component"
    >
      <span
        class="p-avatar-text"
      >
        P
      </span>
    </div>
    <div
      class="p-avatar p-component"
    >
      <span
        class="p-avatar-icon pi pi-search"
      />
    </div>
    <div
      class="p-avatar p-component p-avatar-image"
    >
      <img
        alt="avatar"
        src="user.png"
      />
    </div>
    <div
      class="p-avatar p-component"
    >
      <span
        class="p-avatar-text"
      >
        +2
      </span>
    </div>
  </div>
</div>
`;
